src/library/ResourceUtils/ResourceUtils.ts

```diff
@@ -70,6 +70,12 @@ export class ResourceUtils {
    * @returns array of elements found at the provided path
    */
   getValuesAtResourcePath(resource: any, elementPath: string): any[] {
+    const pathSections = elementPath.split('.');
+    if (!resource || (resource.resourceType !== pathSections[0])) return [];
```

Comment:
how about the second check to be  `elementPath.startsWith(resource.resourceType);`  ?